### PR TITLE
Document tagged reductions

### PR DESCRIPTION
**Pull request by masterleinad**
Related to https://github.com/kokkos/kokkos/pull/4663.

### CODE REVIEW

File: API-Reference/Kokkos::parallel_reduce.md [hunk 1]
```diff
@@ -82,9 +82,9 @@ Kokkos::parallel_reduce(const ExecPolicy& policy,
   * If `ExecPolicy` is `MDRangePolicy` the `functor` has a member function of the form `operator() (const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const` or `operator() (const WorkTag, const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const` 
     * The `WorkTag` free form of the operator is used if `ExecPolicy::work_tag` is not `void`.
     * `N` must match `ExecPolicy::rank`
-  * If the `functor` is a lambda, `ReducerArgument` must satisfy the `Reducer` concept or `ReducerArgumentNonConst` must be a POD type with `operator +=` and `operator =` or a `Kokkos::View`.  In the latter case, the default `Sum` reduction is applied. 
+  * If the `functor` is a lambda, `ReducerArgument` must satisfy the `Reducer` concept or `ReducerArgumentNonConst` must be a POD type with `operator +=` and `operator =` or a `Kokkos::View`.  In the latter case, the default `Sum` reduction is applied. In this case, the potential `init`/`join`/`final` member functions must be untagged even for tagged reductions.
```
Comment (dalg24) [inline API-Reference/Kokkos::parallel_reduce.md:85]: "untagged" does not seem to be defined.
```suggestion
  * If the `functor` is a lambda, `ReducerArgument` must satisfy the `Reducer` concept or `ReducerArgumentNonConst` must be a POD type with `operator +=` and `operator =` or a `Kokkos::View`.  In the latter case, the default `Sum` reduction is applied. If provided, the `init`/`join`/`final` member functions do not take a `WorkTag` argument.
```

**Comment by masterleinad** [comment]
> please update this so that you use the new sphinx branch

I rebased accordingly.

**Comment by fnrizzi** [comment]
@masterleinad 
all editable sources from now on are under `source/`.
I propose this: if you want/can, please *move* your edits to this file: https://github.com/kokkos/kokkos-core-wiki/blob/sphinx_porting_stage1/source/API/core/parallel-dispatch/parallel_reduce.md
or if you don't want/don't have time I can port your edits directly.

**Comment by masterleinad** [comment]
@fnrizzi I moved the changes.

**Review by fnrizzi — APPROVED** [review]

**Comment by fnrizzi** [comment]
@dalg24 let me know if you are ok merging this

**Pull request merged**